### DISCUSSION TITLE
BedrockCohereEmbeddingOptions does not implement EmbeddingOptions

### DISCUSSION THREAD

**Discussion by sharathcshekhar** [spring-projects/spring-ai#381] [Q&A] [answered]
In the documentation to use Cohere embeddings, https://docs.spring.io/spring-ai/reference/api/embeddings/bedrock-cohere-embedding.html#embedding-options, the below code example is provided.

```
EmbeddingResponse embeddingResponse = embeddingClient.call(
    new EmbeddingRequest(List.of("Hello World", "World is big and salvation is near"),
        BedrockCohereEmbeddingOptions.builder()
        	.withInputType(InputType.SEARCH_DOCUMENT)
        .build()));
```
But this code does not work because BedrockCohereEmbeddingOptions does not implement EmbeddingOptions. I am on 0.8.0. Is there something I am missing? Thanks.

**Comment by tzolov** [comment]
Hi @sharathcshekhar, 
Thanks for catching this. It is is indeed a bug and i can reproduce it. Will submit a fix for the 0.8.1-SNAPSHOT branch shortly

**Reply by tzolov** [reply, marked as answer]
FYI, this problem is resolved with https://github.com/spring-projects/spring-ai/commit/5e6815863c8520da8a0b29eb564bcb9978dcc641

**Reply by sharathcshekhar** [reply]
Thanks for the quick fix!
